packages/block-editor/src/components/global-styles/get-global-styles-changes.js

```diff
@@ -176,29 +171,75 @@ function getGlobalStylesChangelist( next, previous ) {
 }
 
 /**
- * From a getGlobalStylesChangelist() result, returns a truncated array of translated changes.
- * Appends a translated string indicating the number of changes that were truncated.
+ * From a getGlobalStylesChangelist() result, returns an array of translated global styles changes, grouped by type.
+ * The types are 'blocks', 'elements', 'settings', and 'styles'.
  *
  * @param {Object}              next     The changed object to compare.
  * @param {Object}              previous The original object to compare against.
  * @param {{maxResults:number}} options  Options. maxResults: results to return before truncating.
- * @return {string[]}                        An array of translated changes.
+ * @return {string[]}                    An array of translated changes.
  */
 export default function getGlobalStylesChanges( next, previous, options = {} ) {
-	const changes = getGlobalStylesChangelist( next, previous );
-	const changesLength = changes.length;
+	let changeList = getGlobalStylesChangelist( next, previous );
+	const changesLength = changeList.length;
 	const { maxResults } = options;
 
-	// Truncate to `n` results if necessary.
-	if ( !! maxResults && changesLength && changesLength > maxResults ) {
-		const deleteCount = changesLength - maxResults;
-		const andMoreText = sprintf(
-			// translators: %d: number of global styles changes that are not displayed in the UI.
-			_n( '…and %d more change', '…and %d more changes', deleteCount ),
-			deleteCount
-		);
-		changes.splice( maxResults, deleteCount, andMoreText );
+	if ( changesLength ) {
+		// Truncate to `n` results if necessary.
+		if ( !! maxResults && changesLength > maxResults ) {
+			changeList = changeList.slice( 0, maxResults );
+		}
+		return Object.entries(
+			changeList.reduce( ( acc, curr ) => {
+				const group = acc[ curr[ 0 ] ] || [];
+				if ( ! group.includes( curr[ 1 ] ) ) {
+					acc[ curr[ 0 ] ] = [ ...group, curr[ 1 ] ];
+				}
+				return acc;
+			}, {} )
+		).map( ( [ key, changeValues ] ) => {
+			const changeValuesLength = changeValues.length;
+			switch ( key ) {
+				case 'blocks': {
+					return sprintf(
+						// translators: %2$s: a list of block names separated by a comma.
+						_n( '%2$s block.', '%2$s blocks.', changeValuesLength ),
+						changeValuesLength,
+						changeValues.join( ', ' )
```

Comment:
```suggestion
						changeValues.join( __( ', ' ) )
```

Ideally, the separator itself should also be localized.

There is a [`wp_get_list_item_separator()` function](https://github.com/WordPress/wordpress-develop/blob/dfd9b1afd2d58c430838a767d7b21524db9c118c/src/wp-includes/l10n.php#L1897-L1907) in the core to get localized separators, but it doesn't currently exist in Gutenberg. There has been an attempt to create such a utility once, but I think an ad hoc approach is fine for now (See [this discussion](https://github.com/WordPress/gutenberg/pull/58256#discussion_r1466398185)).

Reply:
Thanks for the links to the context!